### DISCUSSION TITLE
html-audio-response plugin causing crash?

### DISCUSSION THREAD

**Discussion by arshiya01** [jspsych/jsPsych#3003] [Q&A]
Hi,

When using the html-audio-response plugin on cognition.run, it can cause crashes or slow performance. The plugin documentation suggests saving the data to your server immediately after the trial and deleting the data from jsPsych's data object. However, since cognition.run manages the server-side, I am unclear how to achieve this on cognition.run

Here is a code snippet where the plugin is being used:

```
var first_disclosure = {
  type: JsPsychHtmlAudioResponse,
  stimulus: "Recording..",
  recording_duration: 180000,
  show_done_button: true,
  done_button_label: 'Done',
  on_finish: function(data){
    jsPsych.data.addProperties({first_disclosure_data: data.response})
  }
};
```

Any assistance with saving the recorded data and deleting it from the jsPsych data object when using cognition.run would be greatly appreciated!

**Comment by ChristopheBossens** [comment]
By using the addProperties function in the on_finish function, you are adding the audio response to every trial in your experiment. I would avoid this because in contrast to deleting data you are now actually adding more data to your experiment. (Not sure if the assignment happens by reference or creates a copy, but better safe than sorry :))

Can you check if it works without using the on_finish function?  I have managed to do so in the past and it worked like a charm on cognition.run. You do need a little code to convert the actual audio from the JSON response afterwards. Let me know if you need help with that.

To delete the audio response from the jsPsych object is probably a little bit more tricky. Normally you would have custom code in the on_finish() that sends data to the server and then deletes the data from the jsPsych object. However, I think cognition.run saves the data automatically after the on_finish function is executed, which means your next chance to delete the data would be at the start of the subsequent trial (e.g., in the on_load() function).

**Reply by arshiya01** [reply]
Thank you!! I used addProperties in every plugin to get an organized csv file in the end, but I guess it makes sense that it only adds more data to the experiment. 
I was able to convert base64 to mp3 files. Thanks again :)
